packages/ember-glimmer/tests/integration/components/life-cycle-test.js

```diff
@@ -1378,8 +1381,64 @@ moduleFor('Run loop and lifecycle hooks', class extends RenderingTest {
 
     this.assertText('wat');
   }
+
+  ['@test element availability in `willInsertElement`'](assert) {
+    let ComponentClass = Component.extend({
+      tagName: 'a',
+      customHref: 'http://google.com',
+      attributeBindings: ['customHref:href'],
+      willInsertElement() {
+        assert.ok(this.element, 'should have element');
+        assert.notOk(this.element.parentNode, 'is detached');
+      }
+    });
+
+    let template = `Hello World`;
+
+    this.registerComponent('foo-bar', { ComponentClass, template });
+
+    this.render(`{{foo-bar id="foo"}}`);
+
+    this.assertElement(this.firstChild, {
+      tagName: 'a',
+      attrs: {
+        id: 'foo',
+        href: 'http://google.com',
+        class: classes('ember-view')
+      }
+    });
+  }
+
+  ['@test `willRender` can set before render'](assert) {
+    let ComponentClass = Component.extend({
+      tagName: 'a',
+      customHref: 'http://google.com',
+      attributeBindings: ['customHref:href'],
+      willRender() {
+        assert.notOk(this.element, 'should not have element');
```

Comment:
Probably ✂️ this assertion ? It's already covered by the general lifecycle tests, so this just distracts from the point of the test imo